plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/ComponentComponent.groovy

```diff
@@ -53,35 +57,160 @@ class ComponentComponent
   RepositoryManager repositoryManager
 
   @DirectMethod
-  List<AssetXO> readAssets(final StoreLoadParameters parameters) {
-    return readAssets(parameters.getFilter('repositoryName'), parameters.getFilter('componentId'))
+  PagedResponse<ComponentXO> read(final StoreLoadParameters parameters) {
+    Repository repository = repositoryManager.get(parameters.getFilter('repositoryName'))
+    securityHelper.ensurePermitted(new RepositoryViewPermission(repository, READ))
+
+    if (!repository.configuration.online) {
+      return null
+    }
+
+    def sort = parameters.sort?.get(0)
+    def querySuffix = ''
+    if (sort) {
+      querySuffix += " ORDER BY ${sort.property} ${sort.direction}"
+      if (sort.property == StorageFacet.P_GROUP) {
+        querySuffix += ", ${StorageFacet.P_NAME} ASC,${StorageFacet.P_VERSION} ASC"
+      }
+      else if (sort.property == StorageFacet.P_NAME) {
+        querySuffix += ", ${StorageFacet.P_VERSION} ASC,${StorageFacet.P_GROUP} ASC"
+      }
+    }
+    if (parameters.start) {
+      querySuffix += " SKIP ${parameters.start}"
+    }
+    if (parameters.limit) {
+      querySuffix += " LIMIT ${parameters.limit}"
+    }
+
+    StorageTx storageTx = repository.facet(StorageFacet).openTx()
+    try {
+      def repositories
+      try {
+        repositories = repository.facet(GroupFacet).leafMembers()
+        querySuffix = " GROUP BY ${StorageFacet.P_GROUP},${StorageFacet.P_NAME},${StorageFacet.P_VERSION}" + querySuffix
```

Comment:
When loading components from a group you need to show only one component per group/name/version.
When not a group that is not necessary as there is only one per group/name/version